pkg/testing/tools/artifacts_api.go

```diff
@@ -291,59 +255,12 @@ func checkResponseAndUnmarshal[T any](resp *http.Response) (*T, error) {
 		return nil, fmt.Errorf("%d: %w", resp.StatusCode, ErrBadHTTPStatusCode)
 	}
 
-	respBytes, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("reading response body: %w", err)
-	}
+	d := json.NewDecoder(resp.Body)
 	result := new(T)
-	err = json.Unmarshal(respBytes, result)
-
+	err := d.Decode(&result)
 	if err != nil {
-		return nil, fmt.Errorf("unmarshaling: %w", err)
+		return nil, fmt.Errorf("failed to parse response: %w", err)
 	}
 
 	return result, nil
 }
-
-func (aac ArtifactAPIClient) GetLatestSnapshotVersion(ctx context.Context) (*version.ParsedSemVer, error) {
```

Comment:
this one I am fine with removing as it does not map to an artifact api operation :+1: